File: Difficulty: Easy/Second Largest/second-largest.java
class Solution {
    public int getSecondLargest(int[] arr) {
        // code here
        int largest=-1;
        int prev=largest;
        for(int i=0;i<arr.length;i++) {
            if(arr[i]>largest) {
                prev=largest;
                largest=arr[i];
            }
            else if(prev<arr[i] && arr[i]<largest) {
                prev=arr[i];
            }
        }
        return prev;
    }
}

File: Difficulty: Easy/Second Largest/second-largest.py
class Solution:
    def getSecondLargest(self, arr):
        first=float('-inf')
        second=float('-inf')
        n=len(arr)
        for i in range(n):
            if arr[i]>first:
                second=first 
                first=arr[i]
            if arr[i]>second and arr[i]!=first:
                second=arr[i]
        if second==float('-inf'):
            return -1 
        else:
            return second